flake.nix

```diff
@@ -8,183 +8,152 @@
       url = "github:oxalica/rust-overlay";
       inputs.nixpkgs.follows = "nixpkgs";
     };
-    crane.url = "github:ipetkov/crane";
   };
 
   outputs = {
     self,
     nixpkgs,
-    crane,
     flake-utils,
     rust-overlay,
     ...
-  }:
+  }: let
+    mkHelix = {
+      pkgs,
+      rustPlatform,
+      stdenv,
+      ...
+    }: let
+      # Next we actually need to build the grammars and the runtime directory
+      # that they reside in. It is built by calling the derivation in the
+      # grammars.nix file, then taking the runtime directory in the git repo
+      # and hooking symlinks up to it.
+      grammars = pkgs.callPackage ./grammars.nix {};
+      runtimeDir = pkgs.runCommand "helix-runtime" {} ''
+        mkdir -p $out
+        ln -s ${./runtime}/* $out
+        rm -r $out/grammars
+        ln -s ${grammars} $out/grammars
+      '';
+    in
+      # Currently rustPlatform.buildRustPackage doesn't have the finalAttrs pattern
+      # hooked up. To get around this while having good customization, mkDerivation is
+      # used instead.
+      stdenv.mkDerivation (self: {
+        # START: Reelvalute the below attrs when
+        # https://github.com/NixOS/nixpkgs/pull/354999
+        # or
+        # https://github.com/NixOS/nixpkgs/pull/194475
+        # Are merged.
+
+        # TODO: Probably change to cargoLock
+        cargoDeps = rustPlatform.importCargoLock {
+          lockFile = ./Cargo.lock;
+        };
+
+        nativeBuildInputs = [
+          rustPlatform.rust.rustc # TODO: Remove
+          rustPlatform.rust.cargo # TODO: Remove
+          pkgs.installShellFiles
+          pkgs.git
+        ];
+
+        # TODO: Remove entire attr
+        buildInputs = with rustPlatform; [
+          cargoSetupHook
+          cargoBuildHook
+          cargoInstallHook
+        ];
+
+        # Use Helix's opt profile for the build.
+        # TODO: s/cargoBuildType/buildType
+        cargoBuildType = "opt";
+        # END: Funny attrs to reevaluate
+
+        name = with builtins; (fromTOML (readFile ./helix-term/Cargo.toml)).package.name;
+        src = pkgs.lib.sources.cleanSource ./.;
```

Comment:
Should we also remove `book/` and `docs/` from the sources ? It's not included in the `hx` binary so any changes there are irrelevant and shouldn't be producing rebuilds.

In the same way, we could ignore `README.md`, `CHANGELOG.md`, `logo*.svg`, `screenshot.png` and maybe others